yt_dlp/extractor/rtve.py

```diff
@@ -70,65 +88,86 @@ def _real_initialize(self):
     def _decrypt_url(png):
         encrypted_data = io.BytesIO(base64.b64decode(png)[8:])
         while True:
-            length = struct.unpack('!I', encrypted_data.read(4))[0]
+            length_data = encrypted_data.read(4)
+            length = struct.unpack('!I', length_data)[0]
             chunk_type = encrypted_data.read(4)
             if chunk_type == b'IEND':
                 break
             data = encrypted_data.read(length)
             if chunk_type == b'tEXt':
-                alphabet_data, text = data.split(b'\0')
-                quality, url_data = text.split(b'%%')
-                alphabet = []
-                e = 0
-                d = 0
-                for l in alphabet_data.decode('iso-8859-1'):
-                    if d == 0:
-                        alphabet.append(l)
-                        d = e = (e + 1) % 4
-                    else:
-                        d -= 1
-                url = ''
-                f = 0
-                e = 3
-                b = 1
-                for letter in url_data.decode('iso-8859-1'):
-                    if f == 0:
-                        l = int(letter) * 10
-                        f = 1
-                    else:
-                        if e == 0:
-                            l += int(letter)
-                            url += alphabet[l]
-                            e = (b + 3) % 4
-                            f = 0
-                            b += 1
-                        else:
-                            e -= 1
-
-                yield quality.decode(), url
+                if b'%%' in data:
+                    alphabet_data, text = data.split(b'\0')
+                    quality, url_data = text.split(b'%%')
+                    alphabet = RTVEALaCartaIE._get_alfabet(alphabet_data)
+                    url = RTVEALaCartaIE._get_url(alphabet, url_data)
+                    quality_str = quality.decode()
+                else:
+                    data = bytes(filter(None, data))
+                    alphabet_data, url_data = data.split(b'#')
+                    alphabet = RTVEALaCartaIE._get_alfabet(alphabet_data)
+
+                    url = RTVEALaCartaIE._get_url(alphabet, url_data)
+                    quality_str = ''
+                yield quality_str, url
```

Comment:
Looking at the sites JS I would have said this is a more accurate implementation:
```suggestion
                data = bytes(filter(None, data))
                alphabet_data, _, url_data = data.partition(b'#')
                quality_str, _, url_data = url_data.rpartition(b'%%')
                quality_str = quality_str.decode() or ''
                alphabet = RTVEALaCartaIE._get_alfabet(alphabet_data)
                url = RTVEALaCartaIE._get_url(alphabet, url_data)
                yield quality_str, url
```
Do you have any examples where `%%` is in the data?